lib/spack/spack/database.py

```diff
@@ -640,6 +644,17 @@ def __init__(
 
         self._write_transaction_impl = lk.WriteTransaction
         self._read_transaction_impl = lk.ReadTransaction
+        self._db_version: Optional[vn.ConcreteVersion] = None
+
+    @property
+    def db_version(self) -> vn.ConcreteVersion:
+        if self._db_version is None:
+            raise AttributeError("db version is not yet set")
```

Comment:
I think this should say something about why -- e.g. "version not set -- DB has not been read yet"

Reply:
dde9e088d3e6df2e21ed09fe1f85a99f0268d916